src/main/scala/ing/wbaa/druid/client/DruidAdvancedHttpClient.scala

```diff
@@ -415,20 +417,27 @@ object DruidAdvancedHttpClient extends DruidClientBuilder {
             )
           }
         }
-        .map {
+        .mapAsync(settings.maxOpenRequests) {
           // consider any response with HTTP Code different from StatusCodes.OK as a failure
           case (triedResponse, responsePromise) =>
             triedResponse match {
               case Success(response) if response.status != StatusCodes.OK =>
-                val failure =
-                  Failure(
-                    new HttpStatusException(
-                      response.status,
-                      response.entity.toStrict(responseParsingTimeout).value.flatMap(_.toOption)
+                response.entity
+                  .toStrict(responseParsingTimeout)
+                  .map(Success(_))
```

Comment:
mapping to Success or Failure can be accomplished by simply writing '.map(Try(_))'

Reply:
It is my understanding it _skips_ the `map()` call if the future yields an exception, hence the `recover()` below it.

Reply:
you are absolutely right, never mind.

Reply:
Documented in b5086fe2b9f5fe03337bab6924281ba81ffb3be9